lib/queuePopulator/IngestionProducer.js

```diff
@@ -110,6 +110,56 @@ class IngestionProducer {
         });
     }
 
+    /**
+     * Validate source bucket configuration and aws source credentials:
+     * - credentials provided are accurate
+     * - endpoint provided is accurate
+     * - remote bucket exists
+     * - remote bucket is version enabled
+     * @param {string} bucket - remote bucket name
+     * @param {Logger.newRequestLogger} logger - log object
+     * @param {function} done - callback(error)
+     * @return {undefined}
+     */
+    validateSourceSetup(bucket, logger, done) {
+        this._s3Client.getBucketVersioning({ Bucket: bucket }, (err, res) => {
+            if (err) {
+                const logParams = {
+                    method: 'IngestionProducer.validateRemoteAccess',
+                    remoteBucket: bucket,
+                    zenkoBucket: this._targetZenkoBucket,
+                    error: err,
+                    errorType: err.code,
+                };
+                if (err.code === 'NoSuchBucket') {
+                    logger.error('remote bucket does not exist', logParams);
```

Comment:
the error messages here aren't really providing any more info than the error itself. I could maybe log potential cause of why the user is facing these specific errors?
Or maybe just collapse everything into a single error log and let the error from aws speak for itself